src/main/java/core/basesyntax/strategy/SupplyOperation.java

```
@@ -0,0 +1,12 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.common.FruitTransaction;
+import core.basesyntax.db.Storage;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction fruitTransaction) {
```

`apply` does not validate its input. The project guidance expects handling invalid input (null parameters). Add a null-check for `fruitTransaction` and throw an informative RuntimeException (e.g., `new RuntimeException("FruitTransaction must not be null")`) to fail fast and produce clear errors for tests.